examples/infinite-scrolling/app/routes/index.tsx

```diff
@@ -0,0 +1,78 @@
+import type { LoaderFunction } from "remix";
+import { useFetcher, useLoaderData } from "remix";
+import { useEffect, useState } from "react";
+import InfiniteScroll from "react-infinite-scroller";
+
+type Joke = {
+  id: number;
+  joke: string;
+};
+
+type LoaderData = {
+  results: Joke[];
+  total_pages: number;
+};
+
+export const loader: LoaderFunction = async ({ request }) => {
+  const url = new URL(request.url);
+  const page = Number(url.searchParams.get("page") || "1");
+
+  const res = await fetch(`https://icanhazdadjoke.com/search?page=${page}`, {
+    headers: { Accept: "application/json" },
+  });
+
+  return res.json();
+};
+
+export default function Index() {
+  const { results, total_pages } = useLoaderData<LoaderData>();
+
+  const fetcher = useFetcher();
+
+  const [page, setPage] = useState(1);
+  const [jokes, setJokes] = useState(results);
+  const [canLoadMore, setCanLoadMore] = useState(true);
+
+  const handleLoadMore = () => {
+    if (fetcher.state === "idle" && canLoadMore) {
+      setCanLoadMore(false);
+      fetcher.load(`/?page=${page + 1}`);
+    }
+  };
+
+  useEffect(() => {
+    if (fetcher.data && fetcher.data.results.length > 0) {
+      setJokes((prev) => [...prev, ...fetcher.data.results]);
```

Comment:
The problem with this approach is that you may run out of memory if the user scrolls too far. I have an implementation of infinite scroll that I like, I just never got around to finishing it up. Do you mind taking a look at that and helping it across the finish line? https://github.com/remix-run/remix/tree/kent/infinite-scroll/examples/infinite-scroll

Reply:
Thank you for the feedback and suggestion 😁  Next week, I'll come up with a new solution based on your approach.

Reply:
Thank you! And thanks for the follow-up 👍

Reply:
I refactored the demo to use `react-virtual` and the fake items from your examples which I also added along with 3 more that use `page` instead of the offset. The "alternative" demo is based on the example of infinite scrolling from `react-virtual`'s docs. 

I am not entirely satisfied with my final solution (especially `page/advanced`) and I'm sure it can be improved, but I am a bit stuck in terms of what I should focus on optimising. Any feedback is greatly appreciated and I will gladly try to implement it if I can.